### ISSUE TITLE
Add more context about the doc version

### ISSUE THREAD

**Issue by shawkins** [keycloak/keycloak#33532]
### Description

From #33490 it was not clear to the user of KC 24 that the hostname v2 docs were not applicable - you might only understand that if you start at the doc root https://www.keycloak.org/documentation 

Ideally each individual page should show the version which the docs are applicable to. Better still would be a doc version drop down similar to what openshift and other projects have.

cc @andymunro

**Comment by vmuzikar** [comment]
Hostname v1 and v2 coexisted only in Keycloak 25. With 26 coming really soon and no 25 micro release expected any more, I'd reject this. @shawkins WDYT?

**Comment by shawkins** [comment]
> I'd reject this. @shawkins WDYT?

No, it's about more than just hostname v1 / v2 - it's about letting the users know who navigate in via a search engine or other direct link what version of keycloak the doc page is applicable to - and ideally also a way to navigate to their doc version of choice.

**Comment by vmuzikar** [comment]
@shawkins Thanks for providing more context. The guides always refer to the latest Keycloak version released. And as there's at the moment only a single supported version of Keycloak, stating a version of the guide seems redundant to me.

But I agree it'd be nice to have some sort of archive of the guides with a dropdown to allow selecting the guide's version. We have archive for the other docs (like server admin docs).

CC @ahus1 @stianst

**Comment by shawkins** [comment]
We already do this for other parts of the documentation: https://www.keycloak.org/docs/latest/server_development/ - you see latest in the URL and there's a version 26.0 hover as you look at the doc.

**Comment by vmuzikar** [comment]
@shawkins Yes, that's what I meant by "like server admin docs". :)

**Comment by paul-civitas** [comment]
The page https://www.keycloak.org/server/all-config is extremely useful, as it shows a variety of environment variables, which are crucial for tuning keycloak in a production environment.

However these environment variables change substantially version to version.

It would be a big enhancement if the online documentation could be versioned. As an example of this the kubernetes documentation here: https://kubernetes.io/docs/setup/

There is a "versions" tab on the top left, and it seems to support up to 5 previous versions. Clicking on 1.27 brings us to this page: https://v1-27.docs.kubernetes.io/docs/setup/

Where there is a yellow banner warning saying that this isn't the current version and that the documentation is only relevant to that specific version. This warning is crucial as people who find the page via search engines might think its for a current version when it's actually for an older version.

**Comment by paul-civitas** [comment]
Consolidating on the backend technology that kubernetes uses for its docs might make sense, for the sake of CNCF alignment and not reinventing the wheel.

**Comment by shawkins** [comment]
@paul-civitas I'm not sure what the kubernetes project is using under the covers, but it's fairly straight-forward to keep using github pages, but use branches and submodules to provide a full doc set. That's how I handled it back with the Teiid project https://github.com/teiid/teiid-documents/tree/gh-pages

@vmuzikar @ahus1 @stianst are we open to publishing multiple versions of the guides? Or should I just pursue a banner for the guides to make the version clear?

**Comment by vmuzikar** [comment]
> Or should I just pursue a banner for the guides to make the version clear?

In the upstream, we support only a single Keycloak version at a time. The docs always reflect that – the latest version. But +1, we could be more explicit about that.

**Comment by shawkins** [comment]
> In the upstream, we support only a single Keycloak version at a time.

That doesn't mean community users will automatically follow the latest and greatest. We see a lot of continued usage of older versions.

> But +1, we could be more explicit about that.

Ok I'll start there.

**Comment by shawkins** [comment]
Based upon the comments of @ahus1 on the two recent prs he seems open to providing some additional versions of the guides. 

@ahus1 and @vmuzikar can you confer on the desired direction?

**Comment by ahus1** [comment]
At the moment we're publishing "nightly" and "current" docs. 

I would hope to publish for example the latest patch release of the previous minor release as well, by exchanging "nightly" with "26.3". 

I would be +1 for that change given the value it would add for the community. If there would be a volunteer to implement it, I'd be happy to review it.

**Comment by shawkins** [comment]
Opened https://github.com/keycloak/keycloak-web/issues/660 so that this issue can be closed by the existing pr.
